pkg/store/bucket.go

```diff
@@ -264,12 +264,15 @@ func (s *BucketStore) SyncBlocks(ctx context.Context) error {
 		}()
 	}
 
+	allIDs := map[ulid.ULID]struct{}{}
+
 	err := s.bucket.Iter(ctx, "", func(name string) error {
 		// Strip trailing slash indicating the a directory.
 		id, err := ulid.Parse(name[:len(name)-1])
```

Comment:
not yours, but.. line above `s~// Strip trailing slash indicating the a directory.~// Strip trailing slash indicating a directory.~`